app/models/king.rb

```diff
@@ -1,6 +1,7 @@
 class King < Piece
   def valid_move?(new_x, new_y)
     return false if off_board?(new_x.to_i, new_y.to_i)
+    return false if game.under_attack?(is_white, new_x.to_i, new_y.to_i) # king is not allowed to move into check
```

Comment:
This line is unnecessary, if you look at piece.rb, line 30, I already use the move_to! method to check if a move by ANY piece, not just a king, will leave that player's king in check, and raise an argument error that prevents the move if that is true.

Reply:
Oh, okay! Thanks for pointing this out... It has been a little while since I dove into the code, and I forgot to check move_to! when I was working on this.

Reply:
Okay, I think I understand our issue here. Right now, the can_move_out_of_check? method within the king.rb file checks for valid_move? but it doesn't try move_to!.

Can we DRY up move_to! so it doesn't do the work of both moving the piece and checking for universal valid_move logic and errors?

It would be great if I could invoke all the checks in move_to! without actually moving the piece. If we could do that, then we could reference those checks in can_move_out_of_check? within the king.rb method. Right now its valid_move? is incomplete.